activation/poetdb.go

```diff
@@ -40,34 +41,43 @@ func (db *PoetDb) ValidateAndStorePoetProof(proof types.PoetProof, poetId [types
 		return false, fmt.Errorf("failed to validate poet proof for poetId %x round %d: %v",
 			poetId, roundId, err)
 	}
+	// TODO(noamnelke): validate signature (or extract public key and use for salting merkle hashes)
+
+	return false, nil
+}
+
+func (db *PoetDb) storePoetProof(proof types.PoetProof, poetId [types.PoetIdLength]byte, roundId uint64,
+	signature []byte) error {
 
 	poetProof, err := types.InterfaceToBytes(&proof)
 	if err != nil {
-		return true, fmt.Errorf("failed to marshal poet proof for poetId %x round %d: %v",
+		return fmt.Errorf("failed to marshal poet proof for poetId %x round %d: %v",
 			poetId, roundId, err)
 	}
 
-	// TODO(noamnelke): validate signature (or extract public key and use for salting merkle hashes)
-
 	ref := sha256.Sum256(poetProof)
 
 	batch := db.store.NewBatch()
 	if err := batch.Put(ref[:], poetProof); err != nil {
-		return true, fmt.Errorf("failed to store poet proof for poetId %x round %d: %v",
+		return fmt.Errorf("failed to store poet proof for poetId %x round %d: %v",
 			poetId, roundId, err)
 	}
 	key := makeKey(poetId, roundId)
 	if err := batch.Put(key[:], ref[:]); err != nil {
-		return true, fmt.Errorf("failed to store poet proof index entry for poetId %x round %d: %v",
+		return fmt.Errorf("failed to store poet proof index entry for poetId %x round %d: %v",
 			poetId, roundId, err)
 	}
 	if err := batch.Write(); err != nil {
-		return true, fmt.Errorf("failed to store poet proof and index for poetId %x round %d: %v",
+		return fmt.Errorf("failed to store poet proof and index for poetId %x round %d: %v",
 			poetId, roundId, err)
 	}
-	db.log.Info("stored proof for round %d PoET id %x", roundId, poetId)
+	db.log.Info("stored proof (id: %x) for round %d PoET id %x", ref[:5], roundId, poetId)
+
+	// TODO: REMOVE THE FOLLOWING LINE ⬇⬇⬇
```

Comment:
wtf?!?!?

Reply:
Following a discussion with @zalmen. Let's discuss if you disagree.